kratos/containers/distributed_vector_exporter.h

```diff
@@ -0,0 +1,347 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//
+#if !defined(KRATOS_DISTRIBUTED_VECTOR_EXPORTER_H_INCLUDED )
+#define  KRATOS_DISTRIBUTED_VECTOR_EXPORTER_H_INCLUDED
+
+
+// System includes
+#include <string>
+#include <iostream>
+#include <functional>
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/atomic_utilities.h"
+
+namespace Kratos
+{
+///@addtogroup ApplicationNameApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Provides a DistributedVectorExporter which implements FEM assemble capabilities
+template<class TIndexType=std::size_t>
+class DistributedVectorExporter
+{
+public:
+    ///@name Type Definitions
+    ///@{
+    typedef TIndexType IndexType;
+    typedef int MpiIndexType;
+
+    /// Pointer definition of DistributedVectorExporter
+    KRATOS_CLASS_POINTER_DEFINITION(DistributedVectorExporter);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    template< class TGlobalIndicesVectorType>
```

Comment:
Maybe it's a stupid question, but why templating the constructor? Wouldn't this conflict with the default constructor?

Reply:
OK. I think that now I understand. The template type will be automatically deduced at instantiation time (this doesn't happen in the default constructor case).